service/share/share.go

```diff
@@ -129,8 +130,32 @@ func (s *service) GetShare(ctx context.Context, dah *Root, row, col int) (Share,
 	return nd.RawData()[1:], nil
 }
 
-func (s *service) GetShares(context.Context, *Root) ([][]Share, error) {
-	panic("implement me")
+func (s *service) GetShares(ctx context.Context, root *Root) ([][]Share, error) {
+	eds, err := ipld.RetrieveData(ctx, root, s.dag, rsmt2d.NewRSGF8Codec())
+	if err != nil {
+		return nil, err
+	}
+
+	flattened := ipld.ExtractODSShares(eds)
```

Comment:
This does looping and there is a loop below, while we can do the conversion in one loop only. This way we avoid zalooping.